WeakAuras/Transmission.lua

```diff
@@ -316,37 +270,318 @@ ChatFrame_AddMessageEventFilter("CHAT_MSG_BN_WHISPER_INFORM", filterFunc)
+end)
+
+local checkButtons = {
+  CreateFrame("CHECKBUTTON", "WeakAurasImportAsACopyCheckButton", buttonAnchor, "ChatConfigCheckButtonTemplate"),
+  CreateFrame("CHECKBUTTON", "WeakAurasUpdateAnimateCheckButton", buttonAnchor, "ChatConfigCheckButtonTemplate"),
+  CreateFrame("CHECKBUTTON", "WeakAurasUpdateActionCheckButton", buttonAnchor, "ChatConfigCheckButtonTemplate"),
+  CreateFrame("CHECKBUTTON", "WeakAurasUpdateLoadCheckButton", buttonAnchor, "ChatConfigCheckButtonTemplate"),
+  CreateFrame("CHECKBUTTON", "WeakAurasUpdateConditionsCheckButton", buttonAnchor, "ChatConfigCheckButtonTemplate"),
+  CreateFrame("CHECKBUTTON", "WeakAurasUpdateTriggerCheckButton", buttonAnchor, "ChatConfigCheckButtonTemplate"),
+  CreateFrame("CHECKBUTTON", "WeakAurasUpdateDisplayCheckButton", buttonAnchor, "ChatConfigCheckButtonTemplate"), 
+  CreateFrame("CHECKBUTTON", "WeakAurasUpdateTrivialCheckButton", buttonAnchor, "ChatConfigCheckButtonTemplate"), 
+}
+
+_G[checkButtons[1]:GetName().."Text"]:SetText(L["Make a copy"])
+checkButtons[1].func = function() WeakAuras.UpdateTooltipButtons() end
+local buttonLabels = {
+  nil,
+  L["Animations"],
+  L["Actions"],
+  L["Load Conditions"],
+  L["Conditions"],
+  L["Trigger Settings"],
+  L["Display Settings"],
+  L["Trivial Settings"],
+}
+for i = 2, #checkButtons do 
+  local button = checkButtons[i]
+  button.label = buttonLabels[i]
+end
+
+
+function WeakAuras.UpdateTooltipButtons(data, children, diffs, codes, linesFromBottom)
+  local import, showCode = buttonAnchor.import, buttonAnchor.showCode
+  if data then
+    import.data = data
+    import.children = children
+    import.diffs = diffs
+    showCode.codes = codes
   end
-  if(ItemRefTooltip.WeakAuras_Tooltip_Button) then
-    ItemRefTooltip.WeakAuras_Tooltip_Button:Hide();
+  import:Disable()
+  -- we only want to disable the show code button if in combat, since clicking it will open options
+  showCode:Enable()
+  showCode.codes = showCode.codes or codes
+  if not showCode.codes or #showCode.codes == 0 then
+    showCode:Hide()
+  else
+    showCode:Show()
   end
-  if(ItemRefTooltip.WeakAuras_Tooltip_Button2) then
-    ItemRefTooltip.WeakAuras_Tooltip_Button2:Hide();
+  if InCombatLockdown() then
+    import:SetText(L["In Combat"])
+    showCode:Disable() 
+  elseif WeakAuras.IsImporting() then
+    import:SetText(L["Import in progress"])
+  elseif WeakAurasSaved.import_disabled then
+    import:SetText(L["Import Disabled"])
+  elseif import.data then  -- no need to enable the button if there's nothing to import
+    import:Enable()
+    if import.diffs then
+      -- position checkButtons
+      local tabsChanged = import.diffs.tabsChanged
+      local update_preferences = WeakAurasSaved.update_preferences
+      for i, button in ipairs(checkButtons) do
+        button:Show()
+        if i ~= 1 then
+          if tabsChanged[button] then
+            button:Enable()
+            button:SetChecked(update_preferences and update_preferences[i] or true)
```

Comment:
Making update_preferences index via the int will make future changes to e.g. insert a checkbox pretty hard.